#168292445 Refactor file and folder structure of the codebase

kdaglas opened the pull request:
## Description ##
This PR is to change the folder structuring and file formatting of our codebase

## Type of change ##
Please select the relevant option
- [x] Bugfix(a non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## Necessary documentation? ##
![Screen Shot 2019-09-10 at 10 12 46 AM](https://user-images.githubusercontent.com/34310682/64610654-6dcc1a00-d3d8-11e9-9595-a16189982453.png)

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Related Pivotal Tracker stories ##
[168292445](https://www.pivotaltracker.com/story/show/168292445)

kdaglas committed:
ch(restructure): change the file structure and the code structure

- change the modules being used in the components
- change the structuring of our file system
- change the routing of the components
- eliminate some unused code snippets

[Starts #168292445]

kdaglas committed:
- change module structuring

espoirMur commented on src/app/app.module.ts:
```diff
@@ -6,21 +6,34 @@ import { ToastrModule } from 'ngx-toastr';
 import { AngularFontAwesomeModule } from 'angular-font-awesome';
-import { JwtInterceptor } from './helpers/jwt.interceptor';
-import { ErrorInterceptor } from './helpers/error.interceptor';
+import { JwtInterceptor } from './interceptors/jwt/jwt.interceptor';
```
What do you think about using absolutes imports bro?

kdaglas replied:
we are supposed to use absolute paths... thanks for reminding me, thats what we are to use... let change those real quick @espoirMur

espoirMur reviewed (commented):
Bro just a quick one, which type of import should we use?
Absolute or relative?

JamesMudidi commented on src/app/components/home/home.component.spec.ts:
```diff
@@ -2,7 +2,7 @@ import { RouterTestingModule } from '@angular/router/testing';
 
 import { HomeComponent } from './home.component';
-import { ComponentsModule } from '../../components/components.module';
+import { SharedModule } from '../../modules/shared/shared.module';
```
Since we are using absolute paths, wouldn't this have to change as below?.
```suggestion
import { SharedModule } from 'src/app/modules/shared/shared.module';
```

kdaglas committed:
ch(fix tests): fix the finacial component tests

shemaeric commented:
@kdaglas this is a good job, I would like to point some few concerns:

**1. I have noticed that there is a change in the navbar for the `login` and `signup` buttons.**

<img width="1440" alt="Screen Shot 2019-09-09 at 20 01 19" src="https://user-images.githubusercontent.com/36806576/64557503-f77cd880-d341-11e9-8ec2-6324dc9ac064.png">

**I think these buttons would look better with the existing design that was there**

<img width="1440" alt="Screen Shot 2019-09-09 at 20 03 01" src="https://user-images.githubusercontent.com/36806576/64557707-71ad5d00-d342-11e9-9bf8-0038205976a1.png">

**2. Also, the design for the dropdown menu  on small screens has changed**

<img width="683" alt="Screen Shot 2019-09-09 at 20 45 30" src="https://user-images.githubusercontent.com/36806576/64557979-1af45300-d343-11e9-9ff7-52c68fea754a.png">

**The design that was there**

<img width="707" alt="Screen Shot 2019-09-09 at 20 46 14" src="https://user-images.githubusercontent.com/36806576/64558011-2d6e8c80-d343-11e9-8216-95ee2beeaddb.png">

**3. I have noticed that after logging in, the `navbar` doesn't change accordingly, it waits until you refresh. means I can still see the login and sign up buttons when logged in.**

<img width="1440" alt="Screen Shot 2019-09-09 at 20 15 08" src="https://user-images.githubusercontent.com/36806576/64558137-7de5ea00-d343-11e9-8f08-f6039d9072d9.png">

Niyitangasam commented on src/app/app.module.ts:
```diff
@@ -6,21 +6,34 @@ import { ToastrModule } from 'ngx-toastr';
+import { NoPropertiesComponent } from './components/properties/no-properties/no-properties.component';
+import { FeaturesModule } from './modules/features/features.module';
+import { AuthenticationModule } from './modules/authentication/authentication.module';
+import { PropertyDetailsComponent } from './components/property-details/property-details.component';
```
Great job @kdaglas, however, it looks like you you have forgotten to change all imports to absolutes.

kdaglas replied:
am working on that they are so many sometimes i lose track, but am working on it

Tevinthuku commented:
Hey @kdaglas , nice work, Im getting this error though, right in the home page.
![image](https://user-images.githubusercontent.com/12128153/64589442-cd610000-d3ad-11e9-8b3e-8227408583e2.png)

kdaglas commented:
@Tevinthuku am looking through this issue, give me a few minutes

MBenedicte commented on src/app/app.component.ts:
```diff
@@ -1,10 +1,22 @@
+  //     }
+  //     this.hide = false;
+  //   }));
+  // }
```
Are these comments necessary?

kdaglas replied:
no they are not... let me remove them... thank you

MBenedicte commented on src/app/helpers/tests/spies.ts:
```diff
@@ -24,7 +24,8 @@ export const profileServiceSpy = createSpyObj('ProfileService', [
   'pushProfile',
-  'getDeposits'
+  'getDeposits',
+  // 'userProfile$',
```
I think this also should be removed

kdaglas replied:
yes am onit... thank you

MBenedicte commented on src/app/modules/features/components/payment/payment.module.ts:
```diff
@@ -0,0 +1,34 @@
+//       InputErrorsComponent],
+//
+// })
+// export class PaymentModule { }
```
What happened to this file? Is it no longer used? I think you can remove it instead of commenting on everything

kdaglas replied:
we nolonger use it... i had made sure that we have fewer modules that will hold our components according to the structure i made... thanks for pointing this out... had not seen it

MBenedicte commented on src/app/modules/features/components/profile/financial-information/financial-information.component.spec.ts:
```diff
@@ -0,0 +1,108 @@
+//   //   const el = fixture.debugElement.query(By.css('.current-savings h2')).nativeElement.textContent;
+//   //   expect(el).toEqual(` ${mockDepositsResponse.data.transactions[0].total_amount_paid} `);
+//   // });
+// });
```
The same as this file

kdaglas replied:
this file is required... it needs refactoring so i just commented it out so that i can create a bug fix for it

MBenedicte commented on src/app/modules/features/components/profile/financial-information/financial-information.component.ts:
```diff
@@ -42,7 +43,9 @@ export class FinancialInformationComponent implements OnInit, OnDestroy {
   }
 
   getCardInfo() {
+    // console.log(this.profileService.userProfile$.subscribe())
```
Remove the comment if they are no longer used

kdaglas replied:
thanks... onit

MBenedicte commented on src/app/modules/features/components/profile/financial-information/financial-information.component.ts:
```diff
@@ -42,7 +43,9 @@ export class FinancialInformationComponent implements OnInit, OnDestroy {
   getCardInfo() {
+    // console.log(this.profileService.userProfile$.subscribe())
     this.profileService.userProfile$.subscribe(response => {
+      console.log(response);
```
the same as the console.log, remove it.

MBenedicte commented on src/app/modules/shared/components/navbar/navbar.component.html:
```diff
@@ -56,16 +56,23 @@
+        </button>
+<!--        <button class="btn" type="submit">-->
+<!--          Log in-->
+<!--        </button>-->
```
Remember to remove the comments, if they no longer used

kdaglas replied:
i have worked on this

MBenedicte commented on src/app/modules/shared/components/navbar/navbar.component.scss:
```diff
@@ -2,6 +2,16 @@ a.nav-link.waves-effect.waves-light:hover {
+  width: 100% !important;
+  min-width: 100% !important;
+  height: 82px !important;
+  //position: fixed !important;
```
the same as here

kdaglas replied:
done

MBenedicte commented on src/app/modules/shared/components/navbar/navbar.component.scss:
```diff
@@ -119,30 +129,57 @@ ul.top-menu > li > a:hover .caret {
-  line-height: 18px !important;
-  letter-spacing: normal !important;
+  //line-height: 18px !important;
+  //letter-spacing: normal !important;
```
Delete these comments

kdaglas replied:
thanks... i have done that

MBenedicte commented on src/app/modules/shared/components/navbar/navbar.component.scss:
```diff
@@ -119,30 +129,57 @@ ul.top-menu > li > a:hover .caret {
+  //padding: 6px 12px !important;
+  //position: relative !important;
+  //text-align: center !important;
+  //width: auto !important;
```
Please delete the comments

kdaglas replied:
even this, i have worked on... thanks

MBenedicte commented on src/app/services/http.service.ts:
```diff
@@ -12,28 +12,27 @@ export class HttpService {
+  //   headers: new HttpHeaders({
+  //      'Content-Type': 'application/json'
+  //   })
+  // };
```
Please the comments

kdaglas replied:
this snippet of code is required for the other refactor PR which will be implemented in... so will keep it here for the meantime, but will be worked on in that PR

Jaman-dedy commented on src/app/app.component.spec.ts:
```diff
@@ -1,18 +1,44 @@
+import { RouterTestingModule } from '@angular/router/testing';
+import { AppComponent } from 'src/app/app.component';
+import {configureTestSuite} from 'ng-bullet';
+import {PropertiesComponent} from './components/properties/properties.component';
```
Consider using absolute import in this file please

kdaglas replied:
am working on it

kdaglas replied:
done

Jaman-dedy commented on src/app/modules/authentication/authentication.routing.ts:
```diff
@@ -0,0 +1,46 @@
+import { Routes } from '@angular/router';
+import { LoginComponent } from './components/login/login.component';
```
Absolute import here

kdaglas replied:
thank you this is done.

Jaman-dedy commented on src/app/modules/authentication/components/login/login-form/login-form.component.spec.ts:
```diff
@@ -7,9 +7,7 @@ import { ToastrModule, ToastrService } from 'ngx-toastr';
 import { LocalStorageService } from 'src/app/services/local-storage.service';
 import { LoginService } from 'src/app/services/login/login.service';
-import { localStorageSpy, loginServiceSpy, resetSpies, toastServiceSpy } from '../../../helpers/spies';
-import { LoginHeaderComponent } from '../login-header/login-header.component';
```
Absolute import here @kdaglas

kdaglas replied:
i fixed this too, thanks

Jaman-dedy commented on src/app/modules/authentication/components/login/login.component.spec.ts:
```diff
@@ -6,12 +6,8 @@ import { RouterTestingModule } from '@angular/router/testing';
 import { ToastrModule, ToastrService } from 'ngx-toastr';
 import { LocalStorageService } from 'src/app/services/local-storage.service';
-import { localStorageSpy, toastServiceSpy } from '../../helpers/spies';
+import { localStorageSpy, toastServiceSpy } from '../../../../helpers/tests/spies';
```
Absolute import here

kdaglas replied:
done

Manorlds-Eaglespark commented on src/app/services/local-storage.service.ts:
```diff
@@ -10,7 +10,9 @@ export class LocalStorageService {
       const value = localStorage.getItem(key);
+      // console.log(value);
       return value ? value : fallback;
+      console.log(value);
```
Let's remove this

kdaglas replied:
done... thanks

Manorlds-Eaglespark commented on src/app/services/local-storage.service.ts:
```diff
@@ -10,7 +10,9 @@ export class LocalStorageService {
   get(key: string, fallback: any) {
     if (isPlatformBrowser(this.platformId)) {
       const value = localStorage.getItem(key);
+      // console.log(value);
```
Let's remove this

kdaglas replied:
done thanks

kdaglas committed:
ch(restructure): work on the requests and reviews from the team

Jaman-dedy reviewed (commented):
Great job @kdaglas , I tested the pr locally and it looks some functionalities are not yet working, the login and the signup button don't redirect anywhere, I can't see trending properties..

Manorlds-Eaglespark commented on src/app/app.routing.ts:
```diff
@@ -2,14 +2,21 @@ import { NgModule } from '@angular/core';
+import { AuthGuard } from './guards/auth.guard';
 
 const routes: Routes = [
   {
```
Is the empty bracket necessary, kindly enlight

Tevinthuku replied:
@Manorlds-Eaglespark , I'm not seeing empty brackets, it's just that it hasn't been modified
![image](https://user-images.githubusercontent.com/12128153/64607695-b207ec00-d3d1-11e9-9bca-6b16f642c4a8.png)

kdaglas replied:
there is actually no empty bracket here... maybe i will sync with you on this and you tell me what is happening

Tevinthuku commented:
@kdaglas , what do you think of having the `input-errors folder` in the `shared/components folder` since its also a reusable component

![image](https://user-images.githubusercontent.com/12128153/64607062-643eb400-d3d0-11e9-9374-097a61cba7d0.png)

Manorlds-Eaglespark commented:
Am not able to see any properties on the home page. I think something is missing here.
<img width="1440" alt="Screenshot 2019-09-10 at 13 47 56" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">

Tevinthuku commented:
> Am not able to see any properties on the home page. I think something is missing here.
> <img alt="Screenshot 2019-09-10 at 13 47 56" width="1440" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">

I was facing the same issue, kindly pull the branch and run the server again.

Manorlds-Eaglespark commented:
> > Am not able to see any properties on the home page. I think something is missing here.
> > <img alt="Screenshot 2019-09-10 at 13 47 56" width="1440" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">
> 
> I was facing the same issue, kindly pull the branch and run the server again.

Done that, no changes yet. Let me try again abit later.

Jaman-dedy commented:
> > Am not able to see any properties on the home page. I think something is missing here.
> > <img alt="Screenshot 2019-09-10 at 13 47 56" width="1440" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">
> 
> I was facing the same issue, kindly pull the branch and run the server again.

I've pulled but still the same @Tevinthuku

kdaglas committed:
ch(restructure): work on the requests and reviews from the team

Tevinthuku commented:
> > > Am not able to see any properties on the home page. I think something is missing here.
> > > <img alt="Screenshot 2019-09-10 at 13 47 56" width="1440" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">
> > 
> > 
> > I was facing the same issue, kindly pull the branch and run the server again.
> 
> I've pulled but still the same @Tevinthuku

Wierd 🤔 . It works on my end.. I'm using `npm run start` I've just noticed that if you use the `ng serve -o` command it doesn't work @Jaman-dedy  @Manorlds-Eaglespark

kdaglas commented:
> > > > Am not able to see any properties on the home page. I think something is missing here.
> > > > <img alt="Screenshot 2019-09-10 at 13 47 56" width="1440" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">
> > > 
> > > 
> > > I was facing the same issue, kindly pull the branch and run the server again.
> > 
> > 
> > I've pulled but still the same @Tevinthuku
> 
> Wierd 🤔 . It works on my end.. I'm using `npm run start` I've just noticed that if you use the `ng serve -o` command it doesn't work @Jaman-dedy @Manorlds-Eaglespark

me too, but let me see how that issue is coming about

kdaglas commented:
> @kdaglas , what do you think of having the `input-errors folder` in the `shared/components folder` since its also a reusable component
> 
> ![image](https://user-images.githubusercontent.com/12128153/64607062-643eb400-d3d0-11e9-9374-097a61cba7d0.png)

speaking of this, you are right, let me change that to be in components

Niyitangasam commented:
> > > > > Am not able to see any properties on the home page. I think something is missing here.
> > > > > <img alt="Screenshot 2019-09-10 at 13 47 56" width="1440" src="https://user-images.githubusercontent.com/28473847/64607746-cfd55100-d3d1-11e9-8fad-c624e3c60c6b.png">
> > > > 
> > > > 
> > > > I was facing the same issue, kindly pull the branch and run the server again.
> > > 
> > > 
> > > I've pulled but still the same @Tevinthuku
> > 
> > 
> > Wierd 🤔 . It works on my end.. I'm using `npm run start` I've just noticed that if you use the `ng serve -o` command it doesn't work @Jaman-dedy @Manorlds-Eaglespark
> 
> me too, but let me see how that issue is coming about

I think that we have to use `npm run start` to start it because as I remember  @CryceTruly changed the way our app starts in the other PR of metadata and `ng serve -o` is no longer working.

JamesMudidi commented:
> Hey @kdaglas , nice work, Im getting this error though, right in the home page.
> ![image](https://user-images.githubusercontent.com/12128153/64589442-cd610000-d3ad-11e9-8b3e-8227408583e2.png)

This issue is puzzling since it comes in when tested locally but it's passing on Heroku as seen
[here](https://landville-frontend-pr-58.herokuapp.com/home)

kdaglas commented:
> > Hey @kdaglas , nice work, Im getting this error though, right in the home page.
> > ![image](https://user-images.githubusercontent.com/12128153/64589442-cd610000-d3ad-11e9-8b3e-8227408583e2.png)
> 
> This issue is puzzling since it comes in when tested locally but it's passing on Heroku

so according to @CryceTruly z PR, our `start` command was changed and the `ng serve -o` is no longer functional... so try to use `npm start` or npm run start`... but in the meantime am looking into a better way to have all the commands working

kdaglas commented:
> @kdaglas this is a good job, I would like to point some few concerns:
> 
> **1. I have noticed that there is a change in the navbar for the `login` and `signup` buttons.**
> 
> <img alt="Screen Shot 2019-09-09 at 20 01 19" width="1440" src="https://user-images.githubusercontent.com/36806576/64557503-f77cd880-d341-11e9-8ec2-6324dc9ac064.png">
> 
> **I think these buttons would look better with the existing design that was there**
> 
> <img alt="Screen Shot 2019-09-09 at 20 03 01" width="1440" src="https://user-images.githubusercontent.com/36806576/64557707-71ad5d00-d342-11e9-9bf8-0038205976a1.png">
> 
> **2. Also, the design for the dropdown menu on small screens has changed**
> 
> <img alt="Screen Shot 2019-09-09 at 20 45 30" width="683" src="https://user-images.githubusercontent.com/36806576/64557979-1af45300-d343-11e9-9ff7-52c68fea754a.png">
> 
> **The design that was there**
> 
> <img alt="Screen Shot 2019-09-09 at 20 46 14" width="707" src="https://user-images.githubusercontent.com/36806576/64558011-2d6e8c80-d343-11e9-8216-95ee2beeaddb.png">
> 
> **3. I have noticed that after logging in, the `navbar` doesn't change accordingly, it waits until you refresh. means I can still see the login and sign up buttons when logged in.**
> 
> <img alt="Screen Shot 2019-09-09 at 20 15 08" width="1440" src="https://user-images.githubusercontent.com/36806576/64558137-7de5ea00-d343-11e9-8f08-f6039d9072d9.png">

I have fixed this, I was testing out some functionality and UI implementation but I have fixed this

Tevinthuku commented:
@kdaglas , I've noticed, once a user logs into the app and redirected back to the home-page the navigation doesn't update accordingly 
![image](https://user-images.githubusercontent.com/12128153/64614514-57768c00-d3e1-11e9-90b5-80c5d6a2db2e.png)

unlike the previous version

![image](https://user-images.githubusercontent.com/12128153/64614438-331aaf80-d3e1-11e9-8ac1-10e90b9597cc.png)

Will this be 1 of the bugs that will need fixing?

Rhytah commented on src/app/components/properties/properties.component.spec.ts:
```diff
@@ -33,14 +32,14 @@ describe('PropertiesComponent', () => {
         HttpClientModule,
         NgxSpinnerModule,
-        AuthLayoutModule,
+        // AuthenticationModule,
```
@kdaglas there's another comment that seems to have evaded your attention.

kdaglas replied:
on it... thanks

Rhytah commented on src/app/interceptors/jwt/jwt.interceptor.spec.ts:
```diff
@@ -1,6 +1,6 @@
 import { TestBed } from '@angular/core/testing';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
-import { httpHandlerSpy, httpRequestSpy } from './spies';
+import { httpHandlerSpy, httpRequestSpy } from '../../helpers/tests/spies';
```
@kdaglas we can transform this into an absolute path

kdaglas replied:
on it... thanks

Rhytah commented on src/app/interceptors/error/error.interceptor.spec.ts:
```diff
@@ -1,7 +1,7 @@
 import {HttpClientTestingModule} from '@angular/common/http/testing';
 import {ErrorInterceptor} from './error.interceptor';
-import {httpHandlerSpy, httpRequestSpy, localStorageSpy, routerSpy} from './spies';
+import {httpHandlerSpy, httpRequestSpy, localStorageSpy, routerSpy} from '../../helpers/tests/spies';
```
@kdaglas change this path to absolute as well

kdaglas replied:
done

dorothyas reviewed (approved):
LGTM

JamesMudidi reviewed (approved):
LGTM

kdaglas commented:
> @kdaglas , I've noticed, once a user logs into the app and redirected back to the home-page the navigation doesn't update accordingly
> ![image](https://user-images.githubusercontent.com/12128153/64614514-57768c00-d3e1-11e9-90b5-80c5d6a2db2e.png)
> 
> unlike the previous version
> 
> ![image](https://user-images.githubusercontent.com/12128153/64614438-331aaf80-d3e1-11e9-8ac1-10e90b9597cc.png)
> 
> Will this be 1 of the bugs that will need fixing?

yes this is going to be a bug fix

MBenedicte commented on src/app/app.component.scss:
```diff
@@ -18,4 +18,4 @@ hr.header-line {
   }
-}
\ No newline at end of file
+}
```
Add the empty line

MBenedicte commented on src/styles.scss:
```diff
@@ -4,3 +4,33 @@
+//body {
+//  margin: 0 auto;
+//  width: 75%;
+//}
```
Please remove the comments

MBenedicte commented on src/styles.scss:
```diff
@@ -4,3 +4,33 @@
+p {
+  font-family: 'Airbnb Cereal App Light', Arial, sans-serif;
+  //font-weight:normal;
+  //font-style:normal;
```
Remove the comments

shemaeric commented:
what do you think of putting these file in respective folders, like other services. For auth, local-storage and http services.
<img width="1060" alt="Screen Shot 2019-09-10 at 16 18 56" src="https://user-images.githubusercontent.com/36806576/64622047-fea9f200-d3e6-11e9-9eea-e98b86902e11.png">

kdaglas commented:
> what do you think of putting these file in respective folders, like other services. For auth, local-storage and http services.
> <img alt="Screen Shot 2019-09-10 at 16 18 56" width="1060" src="https://user-images.githubusercontent.com/36806576/64622047-fea9f200-d3e6-11e9-9eea-e98b86902e11.png">

this is being worked in another PR that is refactoring the services @shemaeric

alexxsanya reviewed (commented):
<img width="747" alt="Screenshot 2019-09-10 at 17 50 23" src="https://user-images.githubusercontent.com/21230051/64624572-8e559d80-d3f3-11e9-815c-ed0fb9e3f2d6.png">

@kdaglas well done on this though I have just noticed that the highlighted menu items have a different font size of 0.9rem from the rest which are 1.0rem. I suggest you fix this.

kdaglas commented:
> <img alt="Screenshot 2019-09-10 at 17 50 23" width="747" src="https://user-images.githubusercontent.com/21230051/64624572-8e559d80-d3f3-11e9-815c-ed0fb9e3f2d6.png">
> 
> @kdaglas I just noticed that the highlighted menu items have a different font size of 0.9rem from the rest which are 1.0rem. I suggest you fix this.

This is going to be fixed in another PR the one for the navbar component and the login component

alexxsanya reviewed (approved):
LGTM. @kdaglas well done on this.

Niyitangasam reviewed (approved):
LGTM

bl4ck4ndbr0wn reviewed (approved):
Great work. @kdaglas, love the new folder structure and the separation of concerns. 
LGTM

malep2007 commented:
> <img alt="Screenshot 2019-09-10 at 17 50 23" width="747" src="https://user-images.githubusercontent.com/21230051/64624572-8e559d80-d3f3-11e9-815c-ed0fb9e3f2d6.png">
> 
> @kdaglas well done on this though I have just noticed that the highlighted menu items have a different font size of 0.9rem from the rest which are 1.0rem. I suggest you fix this.

@alexxsanya add an issue so we can keep track of this.
